lib/open_food_network/payments_report.rb

```diff
@@ -41,7 +41,10 @@ def table_items
       return [] unless @render_table
 
       orders = search.result
-      payments = orders.map { |o| o.payments.select(&:completed?) }.flatten # Only select completed payments
+      payments = orders.includes(:payments).map do |order|
+        order.payments.select(&:completed?)
+      end.flatten
+
```

Comment:
I think something like this might be even more efficient:
```ruby
payments = Spree::Payment.completed.where(order_id: orders.select('spree_orders.id'))
```

Reply:
Yes, I also wanted to get rid of this `map` but it was so quick and non-distracting to do this that I thought to improve in a later PR.